--- origsrc/ghex-2.8.2/src/Makefile.am	2004-10-17 13:53:52.000000000 -0500
+++ src/ghex-2.8.2/src/Makefile.am	2006-12-06 22:27:19.841125000 -0600
@@ -22,16 +22,26 @@
 
 libgtkhex_la_SOURCES = \
 	hex-document.c			\
+	hex-document-ui.c			\
 	gtkhex.c			\
 	gtkhex-private.h		\
 	accessiblegtkhex.c		\
 	accessiblegtkhexfactory.c	\
 	accessiblegtkhex.h		\
 	accessiblegtkhexfactory.h	\
+	findreplace.c converter.c ghex.h    \
+	ui.c ghex-window.c ghex-window.h \
+	config.c chartable.c    \
+	preferences.c print.c \
+	gnome-print-font-dialog.c gnome-print-font-dialog.h     \
+	gnome-print-font-picker.c gnome-print-font-picker.h     \
+	hex-dialog.c hex-dialog.h			\
 	$(BUILT_SOURCES)
 
+libgtkhex_la_LDFLAGS = -no-undefined
+
 libgtkhex_la_LIBADD = \
-	$(LIBGTKHEX_LIBS)	\
+	$(GHEX_LIBS)	\
 	$(GAILUTIL_LIBS)
 
 libgtkhex_HEADERS = \
@@ -39,14 +49,8 @@
 	gtkhex.h
 
 ghex2_SOURCES = \
-	hex-document-ui.c preferences.c findreplace.c converter.c	\
-	config.c main.c ui.c chartable.c session.c		\
-	print.c ghex-window.c ghex-window.h				\
-	gnome-print-font-dialog.c gnome-print-font-dialog.h		\
-	gnome-print-font-picker.c gnome-print-font-picker.h		\
-	factory.c factory.h hex-dialog.c hex-dialog.h			\
-	configuration.h preferences.h findreplace.h print.h		\
-	converter.h chartable.h ui.h session.h
+	factory.c factory.h \
+	main.c session.c session.h
 
 ghex-marshal.h: ghex-marshal.list $(GLIB_GENMARSHAL)
 	$(GLIB_GENMARSHAL) $< --header --prefix=ghex_marshal > $@
@@ -61,8 +65,8 @@
 	ghex-marshal.list
 
 LDADD =	\
+	libgtkhex.la	\
 	$(POPT_LIBS)	\
-	$(GHEX_LIBS)	\
-	libgtkhex.la
+	$(GHEX_LIBS)
 
 CLEAN_FILES = $(BUILT_SOURCES)
